pkg/kademlia/kademlia.go

```diff
@@ -23,8 +23,9 @@ import (
 )
 
 const (
-	maxBins        = 16
-	nnLowWatermark = 2 // the number of peers in consecutive deepest bins that constitute as nearest neighbours
+	maxBins         = 16
+	nnLowWatermark  = 2 // the number of peers in consecutive deepest bins that constitute as nearest neighbours
+	maxConnAttempts = 3 // when there is maxConnAttempts failed connect calls for a given peer it is considere non-connectable
```

Comment:
questions about this:
1. if i start my node and i am disconnected from the internet, does this mean that within 3 attempts to connect to peers will be address book get entirely truncated? is this the desirable behavior?
2. how long will it take to reach this limit in the above case?
3. if the connection breaker is hit how long does it take to reach the third attempt?

Reply:
So, first thing is, the PR is still in progress, sorry, so I might change stuff :)
But to answer anyway about the idea I am testing now:
1. So, the conn attempts and connection breaker are not related. The connection breaker should recognize the situation of multiple failed attempts across bee, and the connect calls will return the appropriate error and when to try again, so we don't try for now reason.
2. I left the same logic as before in kademlia, the only difference is that after 3 attemts to the same peer I would remove it from the wait next map and remove it from the addressbook so it is not retried again, unless it is added to the addressbook again somehow and then it will go all over again
3. If the connection breaker is hit, the all connection calls are postponed until returned `tryAfter`, and I am not taking into account those as failed connection attempts, only if the error is not from the connection breaker

Does this makes sense? Maybe I missed something, this was just to try quick fix, I am still testing how it behaves

Reply:
Regarding the release, I asked the same, you can check the linked issue :)

Reply:
This PR would be good to be merged before the release as it would reduce the error messages when connecting to staled nodes greatly. The issue was made with this in mind.